oracle/provider/binance.go

```diff
@@ -339,6 +349,32 @@ func (p *BinanceProvider) GetAvailablePairs() (map[string]struct{}, error) {
 	return availablePairs, nil
 }
 
+// ConfirmPairAvailability takes a list of pairs that are meant to be subscribed
```

Comment:
Now that you've pulled this logic into it's own method it seems like the exact same code for each provider. Do you think there is value in combining these into a single method that accepts a Provider interface?

Reply:
Agreed! I changed it to be its own method in the provider package inside of `available_pairs.go`.